src/classes/speciessite.cpp

```diff
@@ -272,240 +276,212 @@ std::vector<int> SpeciesSite::yAxisAtomIndices() const
 }
 
 // Set target Elements for selection as sites
-bool SpeciesSite::setElements(const std::vector<Elements::Element> &els)
+bool SpeciesSite::setDynamicElements(const std::vector<Elements::Element> &els)
 {
-    elements_.clear();
+    dynamicElements_.clear();
 
-    return std::all_of(els.begin(), els.end(), [&](const auto el) { return addElement(el); });
+    if (std::all_of(els.begin(), els.end(), [&](const auto el) { return addDynamicElement(el); }))
+        return generateUniqueSites();
+    else
+        return false;
 }
 
 // Return elements for selection as sites
-const std::vector<Elements::Element> SpeciesSite::elements() const { return elements_; }
+const std::vector<Elements::Element> SpeciesSite::dynamicElements() const { return dynamicElements_; }
 
 // Add target atom type for selection as sites
-bool SpeciesSite::addAtomType(const std::shared_ptr<AtomType> &at)
+bool SpeciesSite::addDynamicAtomType(const std::shared_ptr<AtomType> &at)
 {
     if (type_ != SiteType::Dynamic)
         return Messenger::error("Setting atom type targets for a non-dynamic site is not permitted.\n");
 
-    if (std::find(atomTypes_.begin(), atomTypes_.end(), at) != atomTypes_.end())
+    if (std::find(dynamicAtomTypes_.begin(), dynamicAtomTypes_.end(), at) != dynamicAtomTypes_.end())
         return Messenger::error("Atom type '{}' is already defined as a target for site '{}'.\n", at->name(), name_);
 
-    atomTypes_.push_back(at);
+    dynamicAtomTypes_.push_back(at);
 
-    return true;
+    return generateUniqueSites();
 }
 
 // Set target atom types for selection as sites
-bool SpeciesSite::setAtomTypes(const std::vector<std::shared_ptr<AtomType>> &types)
+bool SpeciesSite::setDynamicAtomTypes(const std::vector<std::shared_ptr<AtomType>> &types)
 {
-    atomTypes_.clear();
+    dynamicAtomTypes_.clear();
 
-    return std::all_of(types.begin(), types.end(), [&](const auto &at) { return addAtomType(at); });
+    if (std::all_of(types.begin(), types.end(), [&](const auto &at) { return addDynamicAtomType(at); }))
+        return generateUniqueSites();
+    else
+        return false;
 }
 
 // Return atom types for selection as sites
-const std::vector<std::shared_ptr<AtomType>> &SpeciesSite::atomTypes() const { return atomTypes_; }
+const std::vector<std::shared_ptr<AtomType>> &SpeciesSite::dynamicAtomTypes() const { return dynamicAtomTypes_; }
 
 // Return fragment definition
 const NETADefinition &SpeciesSite::fragment() const { return fragment_; }
 
+// Generate unique sites
+bool SpeciesSite::generateUniqueSites()
+{
+    sitesOriginAtomsIndices_.clear();
+    sitesXAxisAtomsIndices_.clear();
+    sitesYAxisAtomsIndices_.clear();
+
+    switch (type_)
+    {
+        case (SiteType::Static):
+        {
+            sitesOriginAtomsIndices_.push_back(staticOriginAtomIndices());
+            sitesXAxisAtomsIndices_.push_back(staticXAxisAtomIndices());
+            sitesYAxisAtomsIndices_.push_back(staticYAxisAtomIndices());
+        }
```

Comment:
Stylistic convention for Dissolve is to not add unnecessary scope with curly braces if not required, but you deffo need a `break;`!
```suggestion
            sitesOriginAtomsIndices_.push_back(staticOriginAtomIndices());
            sitesXAxisAtomsIndices_.push_back(staticXAxisAtomIndices());
            sitesYAxisAtomsIndices_.push_back(staticYAxisAtomIndices());
            break;
```

Reply:
Oops!